opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Wise.kt

```diff
@@ -0,0 +1,563 @@
+package de.geeksfactory.opacclient.apis
+
+import de.geeksfactory.opacclient.i18n.StringProvider
+import de.geeksfactory.opacclient.networking.HttpClientFactory
+import de.geeksfactory.opacclient.networking.NotReachableException
+import de.geeksfactory.opacclient.objects.*
+import de.geeksfactory.opacclient.searchfields.BarcodeSearchField
+import de.geeksfactory.opacclient.searchfields.DropdownSearchField
+import de.geeksfactory.opacclient.searchfields.SearchField
+import de.geeksfactory.opacclient.searchfields.SearchQuery
+import de.geeksfactory.opacclient.utils.jsonObject
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.*
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+import kotlin.experimental.and
+import kotlin.math.floor
+
+data class Authentication(val libraryId: String, val patronSystemId: String,val token: String, val membershipExpiryDate: org.joda.time.LocalDate )
+
+lateinit var apiKeyId: String
+lateinit var apiKey: String
+lateinit var applicationName: String
+var auth: Authentication? =  null
```

Comment:
These leak into the variable scope of the whole package, don't they? This seems unclean and prone to cause confusion. Can we make tie them to the Wise class, please? :)

Reply:
yes will do